### PR TITLE
QuTech SPI rack wrapper classes

### PR DESCRIPTION
**Pull request by Rubenknex**
Added the QuTech SPI rack wrapper classes.

Using these files, a user would have to pip install the package (yet to be uploaded) that contains the low-level code that the instruments use to communicate with the SPI rack modules.

@giulioungaretti @peendebak @CJvanDiepen

### CODE REVIEW

**Comment by Rubenknex** [comment]
@eendebakpt Should we merge this PR so you can continue in another one?

**Comment by giulioungaretti** [comment]
@Rubenknex @eendebakpt  I'd wrap the imports and raise  more descriptive exceptions.

**Comment by Rubenknex** [comment]
@giulioungaretti What exactly do you mean by wrapping the imports? I assume you are referring to the spirack module imports?

File: qcodes/instrument_drivers/QuTech/D4.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from qcodes import Instrument
+
+from spirack import D4_module
```
Comment (giulioungaretti) [inline qcodes/instrument_drivers/QuTech/D4.py]: @Rubenknex I mean:

``` python
try:
     from spirack import D4_module
except ImportError"
    "some helpful message about d4_module " 
```
Does it make sense ?

**Comment by jenshnielsen** [comment]
On a related note can you add the spirack modules to `autodoc_mock_imports` around here https://github.com/QCoDeS/Qcodes/blob/master/docs/conf.py#L388 This allows the autodoc api documentation to be autogenerated even if the spirack modules are not installed (as on travis) Autodoc requires the module to be documented to be importable

File: qcodes/instrument_drivers/QuTech/D4.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+from functools import partial
+
+
+class D4(Instrument):
+    """
+    Qcodes driver for the D4 ADC SPI-rack module.
+    """
```
Comment (giulioungaretti) [inline qcodes/instrument_drivers/QuTech/D4.py]: We may want a proper docstring for the spi_rack and module args. Here.

File: qcodes/instrument_drivers/QuTech/D5a.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    -   set_dacs_zero   set all DACs to zero voltage
+
+    parameters:
+    -   dacN:       get and set DAC voltage
+    -   stepsizeN   get the minimum step size corresponding to the span
+    -   spanN       get and set the DAC span: '4v uni', '4v bi', or '2.5v bi'
+
+    where N is the DAC number from 1 up to 16
+    """
+
```
Comment (giulioungaretti) [inline qcodes/instrument_drivers/QuTech/D5a.py]: And here.

File: qcodes/instrument_drivers/QuTech/S5i.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+from qcodes import Instrument
+from qcodes.utils.validators import Bool, Numbers
+
+from spirack import S5i_module
+
+
+class S5i(Instrument):
+    """
+    Qcodes driver for the S5i RF generator SPI-rack module.
+    """
```
Comment (giulioungaretti) [inline qcodes/instrument_drivers/QuTech/S5i.py]: and here .

File: qcodes/instrument_drivers/QuTech/F1d.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+from qcodes.utils.validators import Enum
+
+from spirack import F1d_module
+
+
+class F1d(Instrument):
+    """
+    Qcodes driver for the F1d IQ-Mixer SPI-rack module.
+    """
+    def __init__(self, name, spi_rack, module, **kwargs):
```
Comment (giulioungaretti) [inline qcodes/instrument_drivers/QuTech/F1d.py]: and here .

File: qcodes/instrument_drivers/QuTech/D4.py (hunk 1 above)
Comment (Rubenknex) [inline qcodes/instrument_drivers/QuTech/D4.py]: Ah yes that was what I thought you meant

**Comment by Rubenknex** [comment]
@peendebak Should the `dac_delay` in D5a.py be a normal `delay` or is there something else?

**Comment by peendebak** [comment]
@Rubenknex In the IVVI driver the parameter is called `dac_delay` so I used that name. `delay` would also be fine

**Comment by peendebak** [comment]
This is solved in #847. The PR can be closed. @jenshnielsen

**Comment by WilliamHPNielsen** [comment]
Superseded by #847 as pointed out by peendebak.

**Pull request closed without merging**